### ISSUE TITLE
Add portable way to unblock/alert blocking curl_multi_wait from another thread

### ISSUE THREAD

**Issue by pps83** [curl/curl#4418]
I'm very happy with curl :) but the way I implemented http code in our project is by using 1ms busy wait worker thread that doesn't even use `curl_multi_wait`. It simply sleeps(1ms) and then does curl_multi_perform. One of the reasons is because I wasn't able to (or rather didn't want to) implement portable code that would be able to "unblock" curl_multi_wait call if I needed to. On non-windows systems generally it's a simple task: an anonymous pipe can be used to alert curl_multi_wait. On windows however, this won't work as select on windows requires socket to be passed. It would be helpful if libcurl instead provided through some api a way to unblock curl_multi_wait call from another thread (perhaps internally it already has something like that?).
On non-windows systems it may implement it as anonymous pipe. On windows it may use self-connected udp socket (this way nobody on localhost would be able to send data to it).

**Comment by bagder** [comment]
Yes it would be useful and I'm sure appreciated by other users too. No, libcurl doesn't have anything like that internally already.

**Comment by pps83** [comment]
Somewhat related quick question: I use CURLOPT_CONNECT_ONLY (for websockets) and when I `curl_easy_cleanup` the handle (after it was remove from multi etc) tcp connection doesn't seem to get closed (according to wireshark). In my case app runs for longer time and I enabled TCP keepalives and a minute after I performed all closing I see keepalive messages in wireshark (1122 is websocket server in nodejs that replies with websocket-close and closes connection):

![image](https://user-images.githubusercontent.com/1614246/65583581-53e01a80-df34-11e9-8fdc-ff153daac85f.png)

Is it curl bug, or I need to dome something else to close connection?

**Comment by pps83** [comment]
I tried to set CURLOPT_FORBID_REUSE and CURLOPT_CONNECT_ONLY, but in this case nothing works: after connect succeeds I cannot make any sends, curl_easy_send weirdly returns CURLE_UNSUPPORTED_PROTOCOL.

**Comment by bagder** [comment]
That sounds like a totally different issue than "unblocking" curl_multi_wait from another thread. If this is a bug, please file this as a separate issue with details to allow us to reproduce.

**Comment by bagder** [comment]
I've jotted down some ideas for [curl_multi_unblock](https://github.com/curl/curl/wiki/curl_multi_unblock) in the wiki. I'm interested in feedback and comments!

**Comment by jay** [comment]
I think an interesting issue is whether a socket always receiving data like in curl_multi_perform would cause an unblock, like can curl_multi_perform be always out (I think I asked about this before my read of the code was unclear on it) and if there is data pending should it return. So say you want to add some URLs to the multi, in your loop you've got a lock free list or something that is checked each time multi perform returns, well you want those transfers added right away, so you in the other thread you add them then you can call unblock?
